trade-finance-manager-ui/server/controllers/team-checking/index.ts

```diff
@@ -0,0 +1,17 @@
+import { Request, Response } from 'express';
+import { TeamToCheck } from '../../helpers/team-checking.helper';
+import { asUserSession } from '../../helpers/express-session';
+import { userIsInTeam } from '../../helpers/user';
+
+type RenderTeamsCheckerProps = {
+  teamCombinations: TeamToCheck[];
+  currentPageTeamRestrictions?: TeamToCheck;
+};
+
+export const renderTeamsChecker = (req: Request, res: Response, { teamCombinations, currentPageTeamRestrictions }: RenderTeamsCheckerProps) => {
+  const { user } = asUserSession(req.session);
+
+  const isUserInTeam = currentPageTeamRestrictions && userIsInTeam(user, currentPageTeamRestrictions.teams);
+
+  return res.render('team-checking/team-checking-page.njk', { currentUserTeams: user.teams, currentPageTeamRestrictions, isUserInTeam, teamCombinations });
```

Comment:
1. Should we use view model pattern here?
2. Can the variable `currentUserTeams` be passed just as a variable.